XMonad/Actions/TreeSelect.hs

```diff
@@ -522,18 +523,22 @@ moveWith f = do
 -- | wait for keys and run navigation
 navigate :: TreeSelect a (Maybe a)
 navigate = gets tss_display >>= \d -> join . liftIO . allocaXEvent $ \e -> do
-    maskEvent d (exposureMask .|. keyPressMask .|. buttonReleaseMask) e
+    -- See XMonad.Prompt Note [Allow ButtonEvents]
+    maskEvent d (exposureMask .|. keyPressMask .|. buttonReleaseMask .|. buttonPressMask) e
+    allowEvents d replayPointer currentTime
 
     ev <- getEvent e
 
-    if ev_event_type ev == keyPress
-      then do
-        (ks, _) <- lookupString $ asKeyEvent e
-        return $ do
-            mask <- liftX $ cleanMask (ev_state ev)
-            f <- asks ts_navigate
-            fromMaybe navigate $ M.lookup (mask, fromMaybe xK_VoidSymbol ks) f
-      else return navigate
+    if | ev_event_type ev == keyPress -> do
+           (ks, _) <- lookupString $ asKeyEvent e
+           return $ do
+               mask <- liftX $ cleanMask (ev_state ev)
+               f <- asks ts_navigate
+               fromMaybe navigate $ M.lookup (mask, fromMaybe xK_VoidSymbol ks) f
+       | ev_event_type ev == buttonPress -> do
+           allowEvents d replayPointer currentTime
```

Comment:
And pls add a comment like [this](https://github.com/xmonad/xmonad-contrib/pull/447/commits/08a165df40acd60b261e141495e0244521c7b86c#diff-83a92be46d4888401e282c9aaa4c8a3d2c761cf52dc5f07d3a1355fe10b27dcfR646) here